SpatialGDK/Source/Private/Interop/GlobalStateManager.cpp

```diff
@@ -29,6 +34,45 @@ void UGlobalStateManager::ApplyData(const Worker_ComponentData& Data)
 	SingletonNameToEntityId = GetStringToEntityMapFromSchema(ComponentObject, 1);
 }
 
+void UGlobalStateManager::ApplyMapData(const Worker_ComponentData& Data)
+{
+	Schema_Object* ComponentObject = Schema_GetComponentDataFields(Data.schema_type);
+	
+	// Set the Map URL.
+	if(Schema_GetObjectCount(ComponentObject, 1) == 1)
+	{
+		SetDeploymentMapURL(GetStringFromSchema(ComponentObject, 1));
+	}
+
+	if(Schema_GetBoolCount(ComponentObject, SpatialConstants::GLOBAL_STATE_MANAGER_ACCEPTING_PLAYERS_ID) == 1)
+	{
+		// Set the AcceptingPlayers state.
+		bool bDataAcceptingPlayers = bool(Schema_GetBool(ComponentObject, SpatialConstants::GLOBAL_STATE_MANAGER_ACCEPTING_PLAYERS_ID));
+
+		if (bDataAcceptingPlayers)
+		{
+			if (bDataAcceptingPlayers != bAcceptingPlayers)
+			{
+				UE_LOG(LogGlobalStateManager, Log, TEXT("GlobalStateManager ApplyMapData - AcceptingPlayers: %s"), bDataAcceptingPlayers ? TEXT("true") : TEXT("false"));
+				bAcceptingPlayers = bDataAcceptingPlayers;
+				AcceptingPlayersChanged.ExecuteIfBound(bAcceptingPlayers);
+			}
+		}
+		else
+		{
+			if (!NetDriver->IsServer())
+			{
+				// TODO: UNR-656 - TLDR: Hack to get around runtime not giving data on streaming queries unless you have write authority.
+				// There is currently a bug in runtime which prevents clients from being able to have read access on the component via the streaming query.
+				// This means that the clients never actually receive updates or data on the GSM. To get around this we are making timed entity queries to
+				// find the state of the GSM and the accepting players. Remove this work-around when the runtime bug is fixed.
+				UE_LOG(LogGlobalStateManager, Warning, TEXT("ApplyMapData - Not yet accepting new players, trying again..."));
+				QueryGSM(true /*bWithRetry*/);
```

Comment:
Super mega nit: Can you put the comment in front of the parameter since that is the convention used elsewhere?